node-tests/e2e/test-support-test.js

```diff
@@ -62,34 +62,90 @@ describe('e2e: provides test support', function() {
     );
   });
 
-  afterEach(async function() {
-    await removeConfig(testProject);
-  });
+  describe('causes tests to fail on CSP violations', async function() {
+    afterEach(async function() {
+      await removeConfig(testProject);
+    });
```

Comment:
Global `afterEach` caused issues because it was running after *every* test even if part of a `describe` hook. Some `describe` blocks are running multiple tests against the same running development server. After fixing live reload support this showed strange timing issues. Moved all tests into describe hooks and avoided global `afterEach` to have a more fine granular control.